tests/test_configurations/test_manager.py

```diff
@@ -235,6 +245,35 @@ def test_agent_attribute_get_set():
     agent_config_manager.verify_private_keys(DUMMY_AEA, lambda x, y, z: None)
 
 
+def test_recursive_updates() -> None:
+    """Test recursive updates."""
+    agent_config_manager = AgentConfigManager.load(DUMMY_AEA, substitude_env_vars=True)
+    agent_config_manager.set_variable(
+        "skills.test_skill.models.scaffold.args.recursive", {"foo": "bar"}
+    )
+    value = cast(
+        Dict,
+        agent_config_manager.get_variable(
+            "skills.test_skill.models.scaffold.args.recursive"
+        ),
+    )
+
+    assert value == {"foo": "bar"}
+
+    agent_config_manager.set_variable(
+        "skills.test_skill.models.scaffold.args.recursive",
+        {"hello": "world"},
+    )
+    value = cast(
+        Dict,
+        agent_config_manager.get_variable(
+            "skills.test_skill.models.scaffold.args.recursive"
+        ),
+    )
+
+    assert value == {"foo": "bar", "hello": "world"}
```

Comment:
This confuses me. Does this mean that we cannot override `dict`s, but only append to them?